auth/email-link.html

```diff
@@ -0,0 +1,236 @@
+          // The link was successfully sent. Inform the user.
+          alert('An email was sent to ' + email + '. Please use the link in the email to sign-in.');
+          // [START_EXCLUDE]
+          // Re-enable the sign-in button.
+          document.getElementById('quickstart-sign-in').disabled = false;
+          // [END_EXCLUDE]
+        }).catch(function(error) {
+          // Handle Errors here.
+          var errorCode = error.code;
+          var errorMessage = error.message;
+          // [START_EXCLUDE]
+          alert('Error: ' + errorMessage);
+          console.log(error);
+          // Re-enable the sign-in button.
+          document.getElementById('quickstart-sign-in').disabled = false;
+          // [END_EXCLUDE]
+        });
+        // [END authwithemail]
+      }
+      document.getElementById('quickstart-sign-in').disabled = true;
+    }
+
+    /**
+     * Handles automatically signing-in the app if we clicked on the sign-in link in the email.
+     */
+    function handleSignIn() {
+      // [START handlesignin]
+      if (firebase.auth().isSignInWithEmailLink(window.location.href)) {
+        // [START_EXCLUDE]
+        // Disable the sign-in button during async sign-in tasks.
+        document.getElementById('quickstart-sign-in').disabled = true;
+        // [END_EXCLUDE]
+
+        // You can also get the other parameters passed in the query string such as state=STATE.
+        // Get the email if available.
+        var email = window.localStorage.getItem('emailForSignIn');
+        if (!email) {
+          // User opened the link on a different device. To prevent session fixation attacks, ask the
+          // user to provide the associated email again. For example:
+          email = window.prompt('Please provide the email you\'d like to sign-in with for confirmation.');
+        }
+        if (email) {
+          // The client SDK will parse the code from the link for you.
+          firebase.auth().signInWithEmailLink(email, window.location.href).then(function(result) {
+            // Clear the URL to remove the sign-in link parameters.
+            if (history && history.replaceState) {
+              window.history.replaceState({}, document.title, window.location.href.split('?')[0]);
+            }
+            // Clear email from storage.
+            window.localStorage.removeItem('emailForSignIn');
+            // Signed-in user's information.
+            var user = result.user;
+            var isNewUser = result.additionalUserInfo.isNewUser;
+            console.log(result)
+          }).catch(function(error) {
+            // Handle Errors here.
+            var errorCode = error.code;
+            var errorMessage = error.message;
+            // [START_EXCLUDE]
+            alert('Error: ' + errorMessage);
+            console.log(error);
+            // Re-enable the sign-in button.
+            document.getElementById('quickstart-sign-in').disabled = false;
```

Comment:
Don't you need to re-enable this on successful sign-in too?

Reply:
Yes, but the successful sign-in will trigger an onAuthStateChanged() which re-enables the button.